### PR TITLE
SAK-32127 If making the Resources tool invisible, hide root

### PR DESCRIPTION
**Pull request by master-bob**
Hiding the page the resources tool is on now sets the root collection
to hidden but users still able to access.

This behavior is allowed by setting the sakai.property
> poh.resources.content.sync.visibility

to true. The default for this property is false.

Some minor cleanup was done in SitePageEditHandler.java since I was in there.

The [main functionality](https://github.com/sakaiproject/sakai/pull/3811/files#diff-a7c10d6ebe7df3b4b7eb94bf638ac846R585) of this ticket was developed & tested against 11.1.

### CODE REVIEW

File: site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java [hunk 1]
```diff
@@ -584,9 +580,24 @@ private boolean pageVisibilityHelper(String pageId, boolean visible, boolean ena
             }
             
             if (saveChanges) {
-                if ( visibility != null ) {
-                    roleConfig.setProperty(PORTAL_VISIBLE, visibility);
+                final Boolean specialHidden = getSitePropertySpecialHidden();
+                if(Boolean.TRUE.equals(specialHidden) && "sakai.resources".equals(toolId)) {
```
Comment (ern) [inline site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java]: overly verbose just use "if (specialHidden && ...)"

File: site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java [hunk 2]
```diff
@@ -584,9 +580,24 @@ private boolean pageVisibilityHelper(String pageId, boolean visible, boolean ena
...
+                    // It is possible, though unlikely, that remove/add property will throw a InUseException or ServerOverloadException
+                    // if so do nothing - ?
+                    try{
+                        if (visibility.equals("true")) {
+                            contentHostingService.removeProperty(siteCollectionId, ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT);
+                        } else {
+                            contentHostingService.addProperty(siteCollectionId, ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT, "true");
+                        }
+                    } catch (InUseException | ServerOverloadException e) {
+                        // do nothing
```
Comment (ern) [inline site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java]: do nothing is fine but you should log it, maybe use level warn, and include the message of the exception in the log.

File: site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java [hunk 3]
```diff
@@ -584,9 +580,24 @@ private boolean pageVisibilityHelper(String pageId, boolean visible, boolean ena
...
             if (saveChanges) {
-                if ( visibility != null ) {
-                    roleConfig.setProperty(PORTAL_VISIBLE, visibility);
+                final Boolean specialHidden = getSitePropertySpecialHidden();
+                if(Boolean.TRUE.equals(specialHidden) && "sakai.resources".equals(toolId)) {
+                    final String siteCollectionId =  contentHostingService.getSiteCollection(placement.getSiteId());
+                    // It is possible, though unlikely, that remove/add property will throw a InUseException or ServerOverloadException
+                    // if so do nothing - ?
+                    try{
+                        if (visibility.equals("true")) {
```
Comment (ern) [inline site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java]: "true".equals(visibility)

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 4]
```diff
@@ -2271,6 +2271,11 @@
 # DEFAULT: true
 # poh.allow.lock=false
 
+# SAK-32127
+# If making the Resources tool Hidden, subsequently make the Resources tool "Hidden but Accessible"
+# DEFAULT: false
+# poh.resources.hidden.special.hidden = true
```
Comment (ern) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties]: this property name makes no sense maybe something like
content.sync.visibility

File: site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java (hunk 1 above)
Comment (master-bob) [inline site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java]: What happens when someone has the property defined as "property=" ?

**Pull request merged**